File: strong.py
import math

def is_strong_number(num):
    """
    Checks if a given number is a strong number.
    """
    original_num = num
    sum_of_factorials = 0
    
    
    if num == 0:
        return False 

    while num > 0:
        digit = num % 10
        sum_of_factorials += math.factorial(digit)
        num //= 10
        
    return sum_of_factorials == original_num

print("Strong numbers from 1 to 5000:")
for i in range(1, 5001):
    if is_strong_number(i):
        print(i)

File: strong.java
public class strong
 {

    
    public static long factorial(int n)
 {
        if (n == 0 || n == 1)
 {
            return 1;
        }
        long fact = 1;
        for (int i = 2; i <= n; i++)
 {
            fact *= i;
        }
        return fact;
    }

    public static boolean isStrong(int number) 
{
        int originalNumber = number;
        long sumOfFactorials = 0;

        while (number > 0) {
            int digit = number % 10;
            sumOfFactorials += factorial(digit);
            number /= 10;
        }
        return sumOfFactorials == originalNumber;
    }

    public static void main(String[] args) 
{
        System.out.println("Strong numbers from 1 to 5000 are:");
        for (int i = 1; i <= 5000; i++) 
{
            if (isStrong(i)) 
{
                System.out.print(i + " ");
            }
        }
    }
}